### PR TITLE
[feature] roaming ghost

### LINKED ISSUE #97: Implement Roaming Ghost

### PR DESCRIPTION
**Pull request by rafiftaris**
### Main changes
- Implemented StageController to store room coordinates
- Implemented ghost roaming by GhostMovement script and NavMeshAgent
- Implemented randomizer at Utils
- Implemented DraggablePointDrawer for scene editor (code is functioning but cluttered and too heavy to run, will refactor in separate issue )

### Tasks List
- [close #97 ]

### Attachments
https://user-images.githubusercontent.com/51603162/152714866-25057256-4dab-41d8-a66a-a18b18e94d7a.mov

### CODE REVIEW

File: Assets/Astralization/Scripts/Stage/StageController.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        for (int i = 0; i < RoomCoordinates.Count; i++)
+        {
+            RoomCoordinate roomCoordinate = RoomCoordinates[i];
+            roomCoordinate.coordinate += transform.position;
+            _roomCoordDict[roomCoordinate.name] = roomCoordinate;
+        }
+        _roomNames = new List<string>(_roomCoordDict.Keys);
+    }
+
+    public static List<string> GetRoomNames()
```
Comment (raihansyahdean) [inline Assets/Astralization/Scripts/Stage/StageController.cs:35]: create setter getter region for all three functions

**Review by raihansyahdean — CHANGES_REQUESTED** [review]
Excellent, just minor add region and documentation for TC and feature please.

**Review by naufalhilmi12345 — APPROVED** [review]
LGTM!

**Review by raihansyahdean — APPROVED** [review]
Excellent!

**Pull request merged**